backend/src/main/java/org/hackbrooklyn/plaza/controller/AnnouncementsController.java

```diff
@@ -42,12 +43,23 @@ public AnnouncementsController(AnnouncementService announcementService) {
     @PostMapping
     public ResponseEntity<Void> addAnnouncement(@AuthenticationPrincipal User user, @RequestBody @Valid AnnouncementBodyRequest reqBody) {
         int id = announcementService.createNewAnnouncement(reqBody.getBody(), user);
-        String link = "/announcements/" + id;
 
+        return new ResponseEntity<>(getLocationHeader("/announcements/" + id), HttpStatus.CREATED);
+    }
+
+    @PreAuthorize("hasAuthority(@authorities.ANNOUNCEMENTS_UPDATE)")
+    @PutMapping
+    public ResponseEntity<Void> updateAnnouncement(@RequestBody @Valid ModifyAnnouncementRequest reqBody) {
+        int id = announcementService.updateAnnouncement(reqBody.getId(), reqBody.getBody());
+
+        return new ResponseEntity<>(getLocationHeader("/announcements/" + id), HttpStatus.NO_CONTENT);
```

Comment:
The `Location` header is meant for 201 Created (and thus for the create announcement endpoint), so we shouldn't return anything other than the status code when updating an announcement.